### ISSUE TITLE
Qwen/Qwen3-Reranker-0.6B memory load and score between 0-1

### ISSUE THREAD

**Issue by snyaggarwal** [OpenConceptLab/ocl_issues#2601]
**Qwen Reranker docs - https://huggingface.co/Qwen/Qwen3-Reranker-0.6B**

### Issues:
1. Scores are not between 0-1.
```
By default, scores are raw logit differences. To get 0-1 probability scores, pass a Sigmoid activation function:

scores = model.predict([(query, doc) for doc in documents], activation_fn=torch.nn.Sigmoid())
```
2. This custom encoder when loaded in memory per request has the potential to crash the service (crashes locally on second request), because every request loads the model again and prev request is not releasing memory cleanly. 

### Solution:
1. use `activation_fn=torch.nn.Sigmoid()` for `Qwen` models
2. custom reranker models should go through a bounded in-process cache instead of being freshly loaded

#### LRU Cache implementation:
- Keeps custom models in a class-level LRU-style cache with a lock.
- Limits custom models to 1 cached model (`settings.RERANKER_CUSTOM_ENCODER_CACHE_SIZE`) by default, so loading Qwen evicts any previously cached custom model before another - large model is loaded.
- Expires idle custom models after 300 seconds (`settings.RERANKER_CUSTOM_ENCODER_CACHE_TTL`) by default.
- Calls eager cleanup (gc.collect(), plus CUDA cache cleanup if applicable) when a custom model is evicted.
